### DISCUSSION TITLE
`--affected` tiggers a pnpm install and is slower than '.'

### DISCUSSION THREAD

**Discussion by tomdavidson** [moonrepo/moon#456] [General]
Hello, when I use the `affected` flag a pnpm install invoked:

```shell
$ moon run root:pre-commit --affected 
▪▪▪▪ pnpm install
Scope: all 10 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
. prepare$ husky install configs/.husky
│ husky - Git hooks installed
└─ Done in 87ms
Done in 2.6s
...
```

Which makes things a bit slow for a precommit githook. When I run the same task without  `--affected` ,  `pnpm install`  is no invoked. Also, with the 28 affected files passed, it takes  4s 203ms but without affected flag it takes 93ms - in both cases there is a 100% cache hit.

**Comment by milesj** [comment]
@tomdavidson If there is a `package.json` or `pnpm-lock.yaml` file that is modified, unfortunately the `pnpm install` is automatically triggered.

Let me think about this. There should be an easy work around.

**Comment by tomdavidson** [comment]
Oh that totally makes sense, but the task is not excluding those files from the input so shouldnt the `pnpm install` still happen when I do not pass ` --affected` ?

**Comment by milesj** [comment]
I added a check for this in this PR: https://github.com/moonrepo/moon/pull/457
